File: JavaScript projects/JavaScript Basics/05.While Loops/Exercises/07_moving.js
function moving(input){
    let availibleSpace = input[0] * input[1] * input[2];
    let i = 3;
    let value = input[i];
    let enoughRoom = true;

    while (value != "Done"){
    
        value = Number(value);
        if (availibleSpace < value){
            enoughRoom = false;
            break;
        }

        availibleSpace -= value;
        i ++;
        value = input[i];
    }
    if (enoughRoom){
        console.log(`${availibleSpace} Cubic meters left.`);
    }else{
        console.log(`No more free space! You need ${value - availibleSpace} Cubic meters more.`);
    }
}
moving(["10","10","2","20","20","20","20","122"]);
moving(["10","1","2","4","6","Done"]);

File: Python projects/Python Basics/05_while_loops/exercises/07_moving.py
apartment_width = int(input())
apartment_length = int(input())
apartment_height = int(input())
total_room = apartment_height * apartment_length * apartment_width
packets = input()

while 0 < total_room and packets != 'Done':
    total_room -= int(packets)
    if total_room < 0:
        continue
    packets = input()

if packets == 'Done':
    print(f"{total_room} Cubic meters left.")
else:
    print(f"No more free space! You need {abs(total_room)} Cubic meters more.")
